### ISSUE TITLE
Anticipating a list results in a generator

### ISSUE THREAD

**Issue by dobesv** [six8/anticipate#5]
When using anticipate for a list of values, it actually provides a generator, not a list.  After the generator is iterated a single time, it appears to be empty.

```
def test_list():
    @anticipate(stringz=[basestring])
    def t(stringz=None):
        assert list(stringz) == list(stringz)

    t([])
    t(['a', 'b', 'c'])
```

This test fails because list() can only consume stringz a single time.

Based on the notation, one would naturally expect to receive a list rather than a generator.  If a generator were desired, some other syntax should be used.

This version of adapt_all resolves the issue:

```
def adapt_all(iter, to_cls):
    """
    Returns a generator that will adapt all objects in an iterable to `cls`
    """
    return [adapt(obj, to_cls) for obj in iter]
```

**Comment by dobesv** [comment]
See also https://github.com/six8/anticipate/pull/6

**Comment by six8** [comment]
Fixed in #8

**Comment by six8** [comment]
The thinking was that a generator would allow you to lazily adapt. So you could conserve memory or CPU if you're only wanting 1 value out of the results instead of all. Of course in practice this hasn't ever been taken advantage of so it makes more sense to just use a list. If we need generator later, we'll add an option to specify you want a generator.
